pkg/comp-functions/functions/common/rbac.go

```diff
@@ -11,7 +11,14 @@ import (
 )
 
 // AddSaWithRole creates a service account with the given policy and binds it to the role.
-func AddSaWithRole(ctx context.Context, svc *runtime.ServiceRuntime, policies []rbacv1.PolicyRule, compName, namespace, suffix string) error {
+// withDeletionProtectionDisabled will add a label to the sa, role and rolbinding's Object to allow deletion.
+func AddSaWithRole(ctx context.Context, svc *runtime.ServiceRuntime, policies []rbacv1.PolicyRule, compName, namespace, suffix string, withDeletionProtectionDisabled bool) error {
```

Comment:
Just an idea: maybe it makes sense to adopt the functional parameters here as well? Or we might end up piling more and more params here.

Reply:
not for now, I want to close that MR not to refactor runtime :D